CDP4-SDK-Samples/SimpleConsoleApp.cs

```diff
@@ -0,0 +1,507 @@
+                "Enter a server's URL for future requests (default is https://cdp4services-test.rheagroup.com, just press Enter):");
+            uri = new Uri(String.IsNullOrWhiteSpace(Console.ReadLine())
+                ? "https://cdp4services-test.rheagroup.com"
+                : Console.ReadLine());
+
+            var dal = new CdpServicesDal();
+            credentials = new Credentials(userName, pass, uri, null);
+            session = new Session(dal, credentials);
+
+            printCommands();
+
+            while (isRunning)
+            {
+                try
+                {
+                    executeCommand(Console.ReadLine());
+                }
+                catch (Exception ex)
+                {
+                    printSeparator();
+                    Console.WriteLine("Something went wrong. Sorry about that.");
+                    Console.WriteLine(ex.Message);
+                    printSeparator();
+                }
+            }
+        }
+
+        private void executeCommand(string command)
+        {
+            var args = command.Trim().Split(" ");
+            switch (args[0])
+            {
+                case openConnection:
+                {
+                    open();
+                    break;
+                }
+                case refreshData:
+                {
+                    refresh();
+                    break;
+                }
+                case reloadData:
+                {
+                    reload();
+                    break;
+                }
+                case openIteration:
+                {
+                    getIteration();
+                    break;
+                }
+                case postPredifinedPerson:
+                {
+                    postPerson();
+                    break;
+                }
+                case postPredifinedParameter:
+                {
+                    postParameter();
+                    break;
+                }
+                case postPfsl:
+                {
+                    postPossibleFiniteStateList();
+                    break;
+                }
+                case postPfslReorder:
+                {
+                    postPossibleFiniteStateListReorder();
+                    break;
+                }
+                case removeParameterFromElementDefinition:
+                {
+                    removeParameter();
+                    break;
+                }
+                case closeConnection:
+                {
+                    close();
+                    break;
+                }
+                case restoreServer:
+                {
+                    restore();
+                    break;
+                }
+                default:
+                {
+                    Console.WriteLine("Unrecognized command.");
+                    break;
+                }
+            }
+        }
+
+        private async Task postPossibleFiniteStateListReorder()
```

Comment:
Do please add simple xml documentation for methods and fields/properties. See othe c# repositories for the general look and feel.